### ISSUE TITLE
Cannot find bazel from mediapipe workspace in Docker

### ISSUE THREAD

**Issue by philiprkk** [google/mediapipe#1972]
I get this error when trying to run the hello world example with Docker on Windows. It seems bazel is unreachable when I'm in the mediapipe workspace, but not when I go up one folder (cd ..). What could be the reason for this? The version of bazel in the  /usr/local/lib/bazel/bin folder appears to be 3.7.2.

> root@01a024bc88dc:/mediapipe# GLOG_logtostderr=1 bazel run --define MEDIAPIPE_DISABLE_GPU=1 mediapipe/examples/desktop/hello_world:hello_world
 (specified in /mediapipe/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.

Bazel binaries for all official releases can be downloaded from here:
  https://github.com/bazelbuild/bazel/releases

You can download the required version directly using this command:
-linux-x86_64)&& chmod +x bazel-3.7.2url -fLO https://releases.bazel.build/3.7.2

**Comment by sgowroji** [comment]
Hi @philiprkk, Did you follow this [documentation](https://google.github.io/mediapipe/getting_started/install.html#installing-using-docker) for installing MediaPipe on Docker. And Could you please share the  steps to  reproduce the above error. Thanks!

**Comment by philiprkk** [comment]
Hi, I did follow that documentation. I install Docker with the WSL 2 integration. I add Ubuntu integration. 
![image](https://user-images.githubusercontent.com/57251565/116970556-80374d80-acb8-11eb-9834-2da9bede7769.png)

I then go through steps 2 and 3 exactly as stated. The output for step 2 doesn't look like what's expected according to the guide. 

`[+] Building 38.9s (8/18)
 => [internal] load build definition from Dockerfile                                                               0.1s
[+] Building 39.0s (8/18)
 => [internal] load build definition from Dockerfile                                                               0.1s
 => => transferring dockerfile: 2.42kB                                                                             0.0s
[+] Building 39.9s (8/18)
 => [internal] load build definition from Dockerfile                                                               0.1s
 => => transferring dockerfile: 2.42kB                                                                             0.0s
 => [internal] load .dockerignore                                                                                  0.0s
[+] Building 390.1s (18/19)
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                      0.0s
 => [ 1/13] FROM docker.io/library/ubuntu:18.04@sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02  4.9s
 => => resolve docker.io/library/ubuntu:18.04@sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02ec  0.0s
 => => sha256:4eb8f7c43909449dbad801c50d9dccc7dc86631e54f28b1a4b13575729065be8 3.32kB / 3.32kB                     0.0s
 => => sha256:01bf7da0a88c9e37ae418d17c0aeed0621524848d80ccb9e38c67e7ab8e11928 26.70MB / 26.70MB                   3.5s
 => => sha256:f3b4a5f15c7a0722b4f22e61b5387317eaf2602c27ffb2bceac9a25f19fbd156 852B / 852B                         0.5s
 => => sha256:57ffbe87baa135002dddb7a7460082c5d6a352186e1be9464c5f31db81378824 188B / 188B                         0.2s
[+] Building 390.2s (19/19) FINISHED
 => [internal] load build definition from Dockerfile                                                               0.1s
 => => transferring dockerfile: 2.42kB                                                                             0.0s
 => [internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 58B                                                                                   0.0s
 => [internal] load metadata for docker.io/library/ubuntu:18.04                                                    1.9s
 => [auth] library/ubuntu:pull token for registry-1.docker.io                                                      0.0s
 => [ 1/13] FROM docker.io/library/ubuntu:18.04@sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02  4.9s
 => => resolve docker.io/library/ubuntu:18.04@sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02ec  0.0s
 => => sha256:4eb8f7c43909449dbad801c50d9dccc7dc86631e54f28b1a4b13575729065be8 3.32kB / 3.32kB                     0.0s
 => => sha256:01bf7da0a88c9e37ae418d17c0aeed0621524848d80ccb9e38c67e7ab8e11928 26.70MB / 26.70MB                   3.5s
 => => sha256:f3b4a5f15c7a0722b4f22e61b5387317eaf2602c27ffb2bceac9a25f19fbd156 852B / 852B                         0.5s
 => => sha256:57ffbe87baa135002dddb7a7460082c5d6a352186e1be9464c5f31db81378824 188B / 188B                         0.2s
 => => sha256:538529c9d229fb55f50e6746b119e899775205d62c0fc1b7e679b30d02ecb6e8 1.41kB / 1.41kB                     0.0s
 => => sha256:16b5bdd2931afed0486ba0ff21bca4b2d09e46f8a8c4b905a9370f96bce2fc52 943B / 943B                         0.0s
 => => extracting sha256:01bf7da0a88c9e37ae418d17c0aeed0621524848d80ccb9e38c67e7ab8e11928                          1.1s
 => => extracting sha256:f3b4a5f15c7a0722b4f22e61b5387317eaf2602c27ffb2bceac9a25f19fbd156                          0.0s
 => => extracting sha256:57ffbe87baa135002dddb7a7460082c5d6a352186e1be9464c5f31db81378824                          0.0s
 => [internal] load build context                                                                                  5.2s
 => => transferring context: 291.59MB                                                                              5.1s
 => [ 2/13] WORKDIR /io                                                                                            0.2s
 => [ 3/13] WORKDIR /mediapipe                                                                                     0.1s
 => [ 4/13] RUN apt-get update && apt-get install -y --no-install-recommends         build-essential         ca  233.2s
 => [ 5/13] RUN pip3 install --upgrade setuptools                                                                  2.0s
 => [ 6/13] RUN pip3 install wheel                                                                                 1.4s
 => [ 7/13] RUN pip3 install future                                                                                2.3s
 => [ 8/13] RUN pip3 install six==1.14.0                                                                           1.5s
 => [ 9/13] RUN pip3 install tensorflow==1.14.0                                                                  127.7s
 => [10/13] RUN pip3 install tf_slim                                                                               1.6s
 => [11/13] RUN ln -s /usr/bin/python3 /usr/bin/python                                                             0.3s
 => [12/13] RUN mkdir /bazel &&     wget --no-check-certificate -O /bazel/installer.sh "https://github.com/bazelb  7.2s
 => [13/13] COPY . /mediapipe/                                                                                     0.4s
 => exporting to image                                                                                             5.3s
 => => exporting layers                                                                                            5.3s
 => => writing image sha256:cd43c5988ef106b20225abbb0e1269baf35aea69de283e1f3f529e902e8aa57d                       0.0s
 => => naming to docker.io/library/mediapipe                                                                       0.0s

C:\Repositories\mediapipe>`

I'm not sure what is going on.

**Comment by sgowroji** [comment]
Hi @philiprkk, Could you please test the above with the latest release and let us know if you still see the above behavior. Thanks!

**Comment by FrederikMindtrace** [comment]
I have the same issue with the currently latest version, following the steps in the [documentation](https://google.github.io/mediapipe/getting_started/install.html#installing-using-docker), so this is not solved. Also, step 3 of the documentation expects there to be a "bazelisk" command, which is not found. Using the steps in the [Hello World page](https://google.github.io/mediapipe/getting_started/hello_world_cpp.html), I encounter the same problem as @philiprkk :

```
root@f58b66b3aeae:/mediapipe# bazel run --define MEDIAPIPE_DISABLE_GPU=1 mediapipe/examples/desktop/hello_world:hello_world
 (specified in /mediapipe/.bazelversion), but it wasn't found in /usr/local/lib/bazel/bin.

Bazel binaries for all official releases can be downloaded from here:
  https://github.com/bazelbuild/bazel/releases

You can download the required version directly using this command:
-linux-x86_64)&& chmod +x bazel-3.7.2url -fLO https://releases.bazel.build/3.7.2
```

**Comment by batkhishign55** [comment]
I am also experiencing the same issue following [this documentation](https://google.github.io/mediapipe/getting_started/install.html#installing-using-docker). Any updates?

**Comment by Roios** [comment]
Same thing here...

**Comment by emad-eldeen** [comment]
I had the same issue. 
solved by installing bazelisk
`apt install nodejs npm
npm install -g @bazel/bazelisk`

**Comment by emad-eldeen** [comment]
The issue turned out to come from files line endings CRLF vs LF

**Comment by Riceball2018** [comment]
There's an issue with the docker file at the Install Bazel section. Either changing the 'b\' to 'b && \' or merging the 2 lines should solve this issue.
![image](https://user-images.githubusercontent.com/42002757/160375017-9d81b349-04a5-4dda-898e-5df5b29b5c6d.png)

**Comment by AdonisGe** [comment]
> The issue turned out to come from files line endings CRLF vs LF
This is the right solution, I use vim to modify the .bazelversion file by "set ff=unix" solving the problem.

**Comment by ychtepenko** [comment]
another quick way to solve the problem: 
sed $'s/\r$//' .bazelversion > .bazelversion

**Comment by Mash2065** [comment]
> > The issue turned out to come from files line endings CRLF vs LF
> 
> This is the right solution, I use vim to modify the .bazelversion file by "set ff=unix" solving the problem.

Yes!Pretty Good！Its Probably because the source code is written on windows, so it cannot be recognized on linux, and vim needs to be converted to unix format.
